### PR TITLE
DR-782 Switch to use the stairway library

### PR DESCRIPTION
**Pull request by ddietterich**
This PR changes TDR to use the Stairway library.

The main change involves the removal of UserRequestInfo from Stairway. Instead, Stairway now allows enumeration of flights by filtering on input parameters. I made the change to put the `subjectId` into all flights as an input. That allows the implementation of the current filtering semantics for job enumeration in JobService.

I had to touch many flights to remove the UserRequestInfo structure, but those changes are easy to spot and don't need detailed review.

TDR still uses waitForFlight(). It is only there for create dataset. I plan to mark it Deprecated in the Stairway interface.

### CODE REVIEW

File: src/test/java/bio/terra/service/dataset/DatasetServiceTest.java [hunk 1]
```diff
@@ -93,16 +99,25 @@ public void setup() {
         // Setup mock sam service
         connectedOperations.stubOutSamCalls(samService);
         flightIdsList = new ArrayList<>();
+        datasetIdList = new ArrayList<>();
     }
 
     @After
     public void teardown() {
+        for (UUID datasetId : datasetIdList) {
+            datasetDao.delete(datasetId);
```
Comment (thathert) [inline src/test/java/bio/terra/service/dataset/DatasetServiceTest.java:109]: What was happening without this before?

**Review by thathert — APPROVED** [review]
`+167 −3,478`, nice! I have one question but otherwise this looks good. Pleasantly surprised at the ease of swapping it out.

File: src/main/java/bio/terra/service/job/JobBuilder.java [hunk 2]
```diff
@@ -33,11 +37,10 @@ public JobBuilder addParameter(String keyName, Object val) {
 
         // check that keyName doesn't match one of the required parameter names
         // i.e. disallow overwriting one of the required parameters
-        boolean isParameterRequired = keyName.equals(JobMapKeys.DESCRIPTION.getKeyName())
-            || keyName.equals(JobMapKeys.FLIGHT_CLASS.getKeyName())
```
Comment (marikomedlock) [inline src/main/java/bio/terra/service/job/JobBuilder.java:37]: Why do we no longer store the flight class in the job parameter map?

File: src/main/java/bio/terra/service/job/JobService.java [hunk 3]
```diff
@@ -123,7 +134,10 @@ void waitForJob(String jobId) {
      */
     public void initialize() {
         try {
-            stairway.initialize(stairwayJdbcConfiguration.getDataSource(), stairwayJdbcConfiguration.isForceClean());
+            stairway.initialize(stairwayJdbcConfiguration.getDataSource(),
+                migrateConfiguration.getDropAllOnStart(),
+                migrateConfiguration.getUpdateAllOnStart());
```
Comment (marikomedlock) [inline src/main/java/bio/terra/service/job/JobService.java:139]: I like having these in the migration configuration instead of in stairway.

**Review by marikomedlock — APPROVED** [review]
Left one question. LGTM.

File: src/test/java/bio/terra/service/dataset/DatasetServiceTest.java (hunk 1 above)
Comment (ddietterich) [reply src/test/java/bio/terra/service/dataset/DatasetServiceTest.java:109]: There are deletes at the bottom of many of the tests, so it was working in the success case. My test didn't have the delete and had a failure first time through, so I added this.

**Review by fboulnois — APPROVED** [review]
lgtm

File: src/main/java/bio/terra/service/job/JobBuilder.java (hunk 2 above)
Comment (ddietterich) [reply src/main/java/bio/terra/service/job/JobBuilder.java:37]: FLIGHT_CLASS was never used anywhere except here.

**Pull request merged**